server/src/main/java/org/spine3/server/entity/storage/ColumnTypeRegistry.java

```
@@ -0,0 +1,133 @@
+     *
+     * <p>If no {@link ColumnType} is found, an {@link IllegalStateException} is thrown.
+     *
+     * @param field the {@link Column} to get type conversion strategy for
+     * @return the {@link ColumnType} for the given {@link Column}
+     */
+    public C get(Column<?> field) {
+        checkNotNull(field);
+
+        Class javaType = field.getType();
+        javaType = Primitives.wrap(javaType);
+        C type = null;
+
+        while (type == null && javaType != null) {
+            type = columnTypeMap.get(javaType);
+            javaType = javaType.getSuperclass();
+        }
+
+        checkState(type != null,
+                   "Could not find storage type for %s.",
+                   field.getType());
+        return type;
+    }
+
+    @VisibleForTesting
+    Map<Class, C> getColumnTypeMap() {
+        return Collections.unmodifiableMap(columnTypeMap);
+    }
+
+    public static <C extends ColumnType> Builder<C> newBuilder() {
+        return new Builder<>();
+    }
+
+    public static <C extends ColumnType> Builder<C> newBuilder(ColumnTypeRegistry<C> src) {
+        final Builder<C> builder = new Builder<>();
+        builder.columnTypeMap.putAll(src.columnTypeMap);
+        return builder;
+    }
+
+    public static class Builder<C extends ColumnType> {
+
+        private final Map<Class, C> columnTypeMap = new HashMap<>();
+
+        private Builder() {
+        }
+
+        /**
+         * Create a mapping between the Java class and the database
+         * {@linkplain ColumnType column type}.
+         *
+         * @param javaType   the Java class to map the value from
+         * @param columnType the the database {@linkplain ColumnType column type} to map value to;
+         *                   must be an instance of {@code <C>}
+         * @param <J>        the Java type
+         * @return self for call chaining
+         */
+        public <J> Builder<C> put(Class<J> javaType, ColumnType<J, ?, ?, ?> columnType) {
```

Let's check for `null`s at least. Or maybe something else as well. 

The aim is to ensure the `ColumnTypeRegistry` is built correctly, or fail fast.